lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -222,6 +222,50 @@ SmallVector<unsigned> getWarpOrder(DistributedEncodingTrait layout,
   return toLinearEncoding(layout, shape).getWarpOrder();
 }
 
+bool isDescendingOrder(triton::gpu::MemDescType type) {
+  auto shape = type.getShape();
+  int rank = shape.size();
+
+  auto ll = triton::gpu::toLinearLayout(shape, type.getEncoding());
+
+  // Use the fastest‐varying input bases to compute the element order
+  auto inDims = ll.getInDimNames();
+  assert(!inDims.empty() && "LinearLayout must have at least one input dim");
+  auto fastestInput = *inDims.begin();
+  const auto &regBases = ll.getBases().lookup(fastestInput);
+
+  // stride[d] = least‐significant (i.e. smallest‐magnitude) contribution.
```

Comment:
It's not clear to me what `stride` you're looking for. Is it the stride of "registers" only?